### ISSUE TITLE
`reportUnreachable` false positive when using `isinstance` to compare `Callable` type against non-final class that does not define a `__call__`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#452]
```py
from typing import Callable

class Foo:
    ...

class Bar(Foo):
    def __call__(self): ...
def foo[T, **P](value: Callable[P, T]):
    if isinstance(value, Foo):
        print(1) # error: unreachable, actually reached at runtime

foo(Bar())
```
upstream issue introduced in 1.1.369. [playground](https://pyright-play.net/?pyrightVersion=1.1.369&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgLABQDAxuQM6tQBiYYAXA1EFQAdKIbM2HAEJEQACm5gAlP3pCoAE2rAoAfV1NSJfXNbUSwFSLH0tO4DwDaAFQA0UAFQeACgF05AG6kAK7UvIRGlDSO3u7OvioCQkg6SKyorDBEKEzUgSHU7oqJaupCCCCoMHIAjEpQAMRQ1CDgIOHBKCDUREwAFlGFUL0wwUZwUN29fdQaw-ggnTBYdIz0DmByMvJKSkA)

**Comment by DetachHead** [comment]
this is now fixed when `strictGenericNarrowing` is enabled (i fixed this by accident, i'm not exactly sure how)
